src/main/java/org/hydev/mcpm/utils/NetworkUtils.java

```diff
@@ -0,0 +1,62 @@
+package org.hydev.mcpm.utils;
+
+import org.apache.hc.client5.http.fluent.Request;
+import org.apache.hc.core5.util.Timeout;
+
+import java.io.IOException;
+import java.util.stream.IntStream;
+
+/**
+ * Network utilities
+ *
+ * @author Azalea (https://github.com/hykilpikonna)
+ * @since 2022-11-01
+ */
+public class NetworkUtils
+{
+    private static final Timeout ONE_S = Timeout.ofSeconds(1);
+    private static final int PING_ITERS = 3;
+
+    static
+    {
+        // Preheat HTTP request module so that subsequent access return similar delay
+        try
+        {
+            Request.head("https://1.1.1.1").connectTimeout(ONE_S).execute();
```

Comment:
Does this have side effects or does it cost a bit? It might be better to shift this onto the first iteration of the ping method if possible (e.g. `lazy static var req = Request.head(...).execute()`, in ping: `discard(req)`).

Reply:
It costs like ~500ms when the class is used for the first time, which does the same thing as you are suggesting.

Reply:
I guess it's okay if the ping method is the only way to interact with NetworkUtils. It kinda sounds like a general class that might have different methods on the future, so forcing 500ms for the first time interact with any component of the class is a bit spooky. I'll leave this up to you whether or not you want to change it.

Reply:
Okay actually I'll initialize it on the first ping call